builtins/web/event/event-target.cpp

```diff
@@ -357,25 +390,25 @@ bool EventTarget::invoke_listeners(JSContext *cx, HandleObject target, HandleObj
 
   // 1. Set event's target to the shadow-adjusted target of the last struct in event's path,
   //   that is either struct or preceding struct, whose shadow-adjusted target is non-null.
+  Event::set_phase(event, Event::Phase::AT_TARGET);
+  Event::set_target(event, target);
   // 2. Set event's relatedTarget to struct's relatedTarget.
+  Event::set_related_target(event, target);
   // 3. Set event's touch target list to struct's touch target list.
   // We only use a single target here as it would appear in a Even#path[0];
   // - shadow adjusted target == target
   // - relatedTarget == target
-  Event::set_phase(event, Event::Phase::AT_TARGET);
-  Event::set_target(event, target);
-  Event::set_related_target(event, target);
 
   // 4. If event's stop propagation flag is set, then return.
-  if (Event::is_stopped(event)) {
+  if (Event::has_flag(event, EventFlag::StopPropagation)) {
     return true;
   }
 
   // 5. Initialize event's currentTarget attribute to struct's invocation target.
   Event::set_current_target(event, target);
   // 6. Let listeners be a clone of event's currentTarget attribute value's event listener list.
   auto list = listeners(target);
-  JS::RootedVector<EventListener> list_clone(cx);
+  JS::RootedVector<ListenerRef> list_clone(cx);
```

Comment:
Very much getting into the weeds of performance optimization here, but using `list_clone.reserve(listeners.length())` would ensure that the list doesn't have to be resized during cloning.